calculator.rb

```diff
@@ -0,0 +1,106 @@
+# acceptable operators
+#this is inelegant I know, I ran out of time!
+operators = [ "\+","\-","\*","\/","divide","DIVIDE","Divide","Div","DIV",
+  "div","add","addition","Add","ADD","ADDITION","Addition",
+  "subtract","Subtract","sub","Sub","SUB","SUBTRACT","mult",
+  "multiply","Multiply","MULTIPLY"
+]
+
+# methods
+def add(num_one, num_two)
+  return convert_input(num_one) + convert_input(num_two)
+
+end
+
+def subtract(num_one, num_two)
+  return convert_input(num_one) - convert_input(num_two)
+
+end
+
+def multiply(num_one, num_two)
+  return convert_input(num_one) * nconvert_input(num_two)
+
+end
+
+def divide(num_one, num_two)
+  return convert_input(num_one)/ convert_input(num_two)
+
+end
+# need to change the reg exp to reject things like punctuation
+def num_check(first_num)
+
+  while first_num =~ /[a-zA-Z]/
+    print "That is not a valid number. Please Enter a number."
+    first_num = gets.chomp
+
+  end
+
+end
+
+def second_num_check(second_num)
+  while second_num =~ /[a-zA-Z]/
+    print "That is not a valid number. Please Enter a number."
+    second_num = gets.chomp
```

Comment:
This method is almost exactly the same as the one above, except for the name of the parameter. That side-steps the big gain that methods give us: being able to use the same routine on different data. You could consolidate both of these into one:

```ruby
def num_check(num)
  while num =~ /[a-zA-Z]/
    print "That is not a valid number. Please Enter a number."
    num = gets.chomp
  end
end
```

It doesn't matter whether we're talking about the first or the second number, the procedure for checking if it's valid is the same.